### PR TITLE
perf(database): speed up Prisma migrations on empty databases

### PR DESCRIPTION
**Pull request by mattinannt**
## What does this PR do?

Running Prisma migrations against a **fresh/empty database** took ~3 minutes. That cost is paid on **every** CI/CD run (the e2e and integration-tests workflows both spin up an empty Postgres) and is a latent production risk: the Docker startup entrypoint (`apps/web/scripts/docker/next-start.sh`) and the Helm migration Job both wrap the same runner in a **300-second timeout** — right at the observed runtime.

**Root cause.** The custom migration runner (`packages/database/src/scripts/migration-runner.ts`) exists to interleave *data* migrations (`migration.ts`, tracked in `DataMigration`) with *schema* migrations (`migration.sql`, tracked by Prisma in `_prisma_migrations`) in strict timestamp order. To do that it spawned a **separate `prisma migrate deploy` subprocess for every unapplied schema migration** — 156 of them on an empty DB, sequentially. The dominant cost was 156 Prisma CLI cold-starts, not the SQL.

**Two changes:**

1. **Batch consecutive schema migrations** — run one `migrate deploy` per contiguous run of schema migrations (the boundaries being data migrations) instead of one per migration. `migrate deploy` is idempotent, so this is equivalent but far cheaper: **156 → 17** deploy spawns on an empty DB, **0** on a fully-applied DB. Interleaving is preserved exactly (consecutive schema migrations are by definition not separated by a data migration). A single up-front `_prisma_migrations` query replaces the 156 per-migration lookups.

2. **Fresh-database fast path** — on a brand-new DB there is no data, so every data migration is a guaranteed no-op (several older ones also reference columns/tables that later schema migrations drop/rename, so they can't even execute against the final schema). Apply the **entire schema history in a single `migrate deploy`** and **baseline** the 24 data migrations (record them applied without running): **17 → 1** deploy spawn, and 24 data-migration transactions skipped. The trigger is resume-safe — if a run is interrupted between the schema deploy and the atomic baseline insert, the next run completes it rather than running data migrations against the final schema.

**Net effect:** empty-DB migration goes from **~3 min → ~6s** (verified locally against Postgres 16 + pgvector).

Also switches the e2e / integration-tests workflows from `pnpm db:migrate:dev` (which rebuilds + regenerates the database package before migrating) to a new `db:migrate:ci` script that runs the already-built runner directly, dropping a redundant rebuild those jobs were doing.

This relies on a load-bearing invariant — **data migrations must be no-ops on an empty database** (seed essential data via the seed script, never a data migration) — now documented in the database README and the `generate-data-migration` template.

## QA / Test Plan

**How to test**

- [ ] Empty DB → `pnpm --filter @formbricks/database build` then run the runner (`node packages/database/dist/scripts/apply-migrations.js` with `DATABASE_URL` set) → **1** `migrate deploy` runs, 0 data migrations execute, log reports "Fresh database detected" + "Baselined 24 data migration(s)"; afterwards `_prisma_migrations` has 156 finished rows and `DataMigration` has 24 rows all `status='applied'`.
- [ ] Re-run against the now-migrated DB → no-op: 0 deploys, all 24 data migrations reported already applied, exit 0.
- [ ] Existing/partially-migrated DB (has recorded data migrations) → takes the interleaved batched path: only contiguous schema segments containing an unapplied migration deploy, data migrations run in timestamp order.
- [ ] Interrupted-baseline recovery: with full schema applied but `DataMigration` emptied, re-run → fast path re-detects and baselines without running data migrations (no error), rather than failing against the final schema.
- [ ] CI: the `e2e` and `integration-tests` workflows apply migrations against a fresh Postgres via `pnpm --filter=@formbricks/database db:migrate:ci` and pass.

**Preconditions / test data**

- A reachable Postgres with the `vector` extension available (CI uses the `pgvector/pgvector` image). No seed data required; the interesting cases are the empty DB and the fully-applied DB.

**Risks & regressions**

- Migration ordering / interleaving correctness — mitigated: consecutive schema migrations are never separated by a data migration, so batching cannot reorder; the fresh path only skips data migrations when there is provably no data (all 24 audited as no-ops on empty tables).
- Fresh-path skip of data migrations assumes they are no-ops on an empty DB. A *future* data migration that seeds essential data would be silently skipped on fresh installs — guarded by documentation (README + data-migration template), per the seed-vs-migrate separation.
- CI `db:migrate:ci` depends on the database package being built by the preceding build step (it is, in both workflows).

**Migrations / env / cutover**

- No new/changed env vars, no new DB migrations, no schema changes. Only the mechanism by which existing migrations are applied changes. Adds one package script (`db:migrate:ci`).

### Required

- [x] Filled out the "QA / Test Plan" section in this PR (behavior, edge cases, preconditions, risks, migrations/env)
- [x] Read How we Code at Formbricks
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand bits
- [x] Ran `pnpm build`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues (no UI changes)

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR (n/a — no UI changes)
- [x] Updated the Formbricks Docs if changes were necessary (database package README updated)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01QQdEAsbHqX6duz4Ge7UXcs

---
_Generated by [Claude Code](https://claude.ai/code/session_01QQdEAsbHqX6duz4Ge7UXcs)_

### CODE REVIEW

File: packages/database/src/scripts/migration-runner.ts [hunk 1]
```diff
@@ -62,125 +62,301 @@ const resolvePrismaBin = async (): Promise<string> => {
...
-      // Record migration failure
-      logger.error(error, `Data migration ${migration.name} failed`);
-      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- we need to check if the migration has a lock
-      if (hasLock) {
-        // Mark migration as failed
-        await prisma.$queryRaw`
+        logger.info(`Data migration ${migration.name} completed successfully`);
+      },
+      { timeout: TRANSACTION_TIMEOUT }
+    );
```
[the remark this answers is not in the document]
Comment (mattinannt) [reply packages/database/src/scripts/migration-runner.ts:335]: Addressed in 9e48663. The `DataMigration` status bookkeeping — the `SELECT`, the `pending` `INSERT`, and the `applied` `UPDATE` — now runs on the interactive-transaction client `tx` instead of the outer `prisma` client, so the "applied" marker commits atomically with the changes `migration.run` makes via `tx` (a rollback can no longer leave a migration recorded as applied while its data is discarded). The post-rollback failure handler intentionally stays on `prisma`, since it must persist after the transaction has already rolled back.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by pandeymangg — APPROVED** [review]
LGTM! 🚀 
also the change from `prisma` to `tx` in the actual runner function would fix a pre existing bug 🙈
